R/RtGam.R

```diff
@@ -70,9 +84,11 @@
 #' )
 #' fit <- RtGam(cases, reference_date)
 #' fit
+# nolint end: line_length_linter
 RtGam <- function(cases,
                   reference_date,
                   group = NULL,
+                  day_of_week = TRUE,
```

Comment:
did you explore to see if you could get rid of the boolean to vector by setting the default to be `day_of_week = reference_date`?

Reply:
I don't think this works because we also want to be able to turn the day of week effect off, so we need a FALSE.

If we were to instead do reference-date/custom-level/NULL it's ends up with the same branching logic, but with slightly different classes.

Reply:
So i push push back if its either vector or NULL it seems clearer but I don't feel that strongly.

Reply:
i.e you can see that day of week is reference date if its written as such in the code whereas currently the logic is quite nested